sdk/provisioning/Azure.Provisioning/src/resourcemanager/Tenant.cs

```diff
@@ -24,7 +24,7 @@ public class Tenant : Resource<TenantData>
         /// <param name="scope">The scope the tenant belongs to.</param>
         /// <param name="tenantId">The tenant id.</param>
         public Tenant(IConstruct scope, Guid? tenantId = null)
-            : base(scope, null, GetName(), ResourceTypeName, "2022-12-01", ResourceManagerModelFactory.TenantData(
+            : base(scope, null, tenantId.HasValue ? tenantId.Value.ToString() : GetName(), ResourceTypeName, "2022-12-01", ResourceManagerModelFactory.TenantData(
```

Comment:
can we move the if/else to GetName and pass the tenantId in so the logic is shared?

Reply:
tenantId requires a Guid, whereas resourceName is a string, so attempting to share the logic would result in double parsing for one of the arguments.

Reply:
nevermind..